### PR TITLE
Allow to close modals

### LINKED ISSUE #345: Improve closing of popins windows
When creating a new event, by default the focus is on the title edit field. If you click the "more options" button, the focus is lost. Consequently, at that time, you can't use the Escape key anymore to dismiss the dialog. (1.0.19).

### PR DESCRIPTION
**Pull request by frankrousseau**
This PR fixes #345 by:

* adding cancel/close buttons to modal.
* handling properly escape key to close modals.

### CODE REVIEW

File: client/app/views/settings_modal.coffee [hunk 1]
```diff
@@ -48,14 +55,15 @@ module.exports = class SettingsModals extends BaseView
...
-        # escape from outside a datetimepicker
-        @close() if e.which is 27 and not e.isDefaultPrevented()
+    hideOnEscape: (event) =>
+        @close() if 27 in [event.which, event.keyCode]
 
 
     # Close the modal.
-    close: -> @$el.modal 'close'
+    close: ->
+        @$el.modal 'hide'
```
Comment (jsilvestre) [inline client/app/views/settings_modal.coffee]: I haven't checked the API. Is the change from "close" to "hide" something you wanted or is it a typo? Just to be sure!

**Comment by jsilvestre** [comment]
Since nobody reviewed it, I did it. I only have one thing I'd like you to clarify, to make sure a typo didn't slip into the code. Then I will merge, and test it to move towards publication.

File: client/app/views/settings_modal.coffee (hunk 1 above)
Comment (frankrousseau) [inline client/app/views/settings_modal.coffee]: It's not a typo. That wrong name made the modal unclosable. It's part of my fix!

**Comment by frankrousseau** [comment]
Thank you @jsilvestre for the review!

**Pull request merged**